pkg/registry/core/node/storage/storage.go

```diff
@@ -104,21 +108,7 @@ func NewStorage(optsGetter generic.RESTOptionsGetter, kubeletClientConfig client
 
 	// Build a NodeGetter that looks up nodes using the REST handler
 	nodeGetter := client.NodeGetterFunc(func(ctx context.Context, nodeName string, options metav1.GetOptions) (*v1.Node, error) {
-		obj, err := nodeREST.Get(ctx, nodeName, &options)
-		if err != nil {
-			return nil, err
-		}
-		node, ok := obj.(*api.Node)
-		if !ok {
-			return nil, fmt.Errorf("unexpected type %T", obj)
-		}
-		// TODO: Remove the conversion. Consider only return the NodeAddresses
-		externalNode := &v1.Node{}
-		err = k8s_api_v1.Convert_core_Node_To_v1_Node(node, externalNode, nil)
-		if err != nil {
-			return nil, fmt.Errorf("failed to convert to v1.Node: %v", err)
-		}
-		return externalNode, nil
+		return coreClient.Nodes().Get(nodeName, options)
```

Comment:
hmmm.... I think this is making a questionable value passing more obvious.  when you change the signature we'll see usage.